File: rust/examples/signer_options/credentials_file.rs
use dotenv::from_filename;
use near_api::prelude::{AccountId, NearToken, NetworkConfig, Signer, Tokens};

#[tokio::main]
async fn main() {
    from_filename("../.env").unwrap();
    let account_id_string = std::env::var("ACCOUNT_ID").unwrap();

    let account_id: AccountId = account_id_string.parse().unwrap();

    // Create a signer via the credentials directory
    let credentials_path = std::path::PathBuf::from("../credentials-file.json"); // Path relative to the root directory of the project
    let signer = Signer::new(Signer::access_keyfile(credentials_path).unwrap()).unwrap(); // Create the signer

    let network = NetworkConfig::testnet();

    // Test the signer by transferring NEAR
    Tokens::of(account_id.clone()) // example-account.testnet
        .send_to("receiver-account.testnet".parse().unwrap())
        .near(NearToken::from_near(1))
        .with_signer(signer.clone())
        .send_to(&network)
        .await
        .unwrap()
        .assert_success();
}

File: javascript/examples/keystore-options/credentials-file.js
import { Account } from "@near-js/accounts";
import { JsonRpcProvider } from "@near-js/providers";
import { KeyPairSigner } from "@near-js/signers";
import { NearToken } from "@near-js/tokens";

const NEAR = new NearToken();

import dotenv from "dotenv";
import fs from "fs";

// Load environment variables
dotenv.config({ path: "../.env" });

// Fetch the private key from a credentials file
const credentialsPath = "../credentials-file.json"; // Path relative to the working directory
const credentials = JSON.parse(fs.readFileSync(credentialsPath));

// Create a signer from the private key
const signer = KeyPairSigner.fromSecretKey(credentials.private_key);

// Create a connection to testnet RPC
const provider = new JsonRpcProvider({
  url: "https://test.rpc.fastnear.com",
});

// Instantiate the account
const accountId = process.env.ACCOUNT_ID;
const account = new Account(accountId, provider, signer);

// Test the signer by transferring NEAR
const sendTokensResult = await account.transferToken(
  NEAR,
  NEAR.toUnits("0.1"),
  "receiver-account.testnet"
);

console.log(sendTokensResult);
